### PR TITLE
Cleanup source, fix demo, add `p` suffix to resolution labels

### PR DESCRIPTION
**Pull request by bikegriffith**

### CODE REVIEW

**Comment by bikegriffith** [comment]
Still needs work.  Testing with video.js v7.0.2 and video-js-contrib-quality-levels v2.0.4 -- selecting a new level does not actually switch to the target resolution. Also seeing artifacts of duplicate levels in the Big Buck Bunny HLS -- these should be de-duped in the level selector.

File: dist/videojs-http-source-selector.cjs.js [hunk 1]
```diff
@@ -67,35 +67,32 @@ var SourceMenuItem = function (_MenuItem) {
...
-      } else if (i == this.options_.index) {
-        this.player().qualityLevels()[i].enabled = true;
+
+    var levels = this.player().qualityLevels();
+    for (var i = 0; i < levels.length; i++) {
+      if (selected.index == levels.length) {
+        // If this is the Auto option, enable all renditions for adaptive selection
+        levels[i].enabled = true;
+      } else if (selected.index == i) {
+        levels.enabled = true;
```
Comment (jfujita) [inline dist/videojs-http-source-selector.cjs.js]: Missing levels[i] array index.

File: dist/videojs-http-source-selector.cjs.js [hunk 2]
```diff
@@ -67,35 +67,32 @@ var SourceMenuItem = function (_MenuItem) {
...
       } else {
-        this.player().qualityLevels()[i].enabled = false;
+        levels.enabled = false;
```
Comment (jfujita) [inline dist/videojs-http-source-selector.cjs.js]: Missing levels[i] array index.

**Comment by jfujita** [comment]
@bikegriffith Thanks for the contributions.  I patched the missing array syntax and the sorting/deduplication works great.  A versioned patch was pushed to [npm](https://www.npmjs.com/package/videojs-http-source-selector)

**Pull request merged**